src/libAtomVM/mailbox.c

```diff
@@ -292,15 +296,19 @@ bool mailbox_peek(Context *c, term *out)
 
     TRACE("Pid %i is peeking 0x%lx.\n", c->process_id, m->message);
 
-    if (c->e - c->heap_ptr < m->body.normal.msg_memory_size) {
-        // ADDITIONAL_PROCESSING_MEMORY_SIZE: ensure some additional memory for message processing, so there is
-        // no need to run GC again.
-        if (UNLIKELY(memory_gc(c, context_memory_size(c) + m->body.normal.msg_memory_size + ADDITIONAL_PROCESSING_MEMORY_SIZE) != MEMORY_GC_OK)) {
+    Message *normal_message = CONTAINER_OF(m, Message, base);
```

Comment:
I did introduce this name of `normal_message` but it looks weird here.

Reply:
I had the same feeling, let's call this `data_message` as opposed to signaling messages (the other option was regular message).